app.js

```diff
@@ -0,0 +1,69 @@
+const STORAGE_KEY = "tasks";
+
+function saveTasks(tasks) {
+  localStorage.setItem(STORAGE_KEY, JSON.stringify(tasks));
+}
+
+function loadTasks() {
+  const json = localStorage.getItem(STORAGE_KEY);
+  return json ? JSON.parse(json) : [];
+}
+
+function renderTasks() {
+  const tasks = loadTasks();
+  const board = document.getElementById("task-board");
+  board.innerHTML = "";
+
+  for (const task of tasks) {
+    const card = document.createElement("div");
+    card.className = "task-card";
+    card.innerHTML = `
+      <p>${task.text}</p>
+      <small>${task.date} <br> ${task.time}</small>
+      <span class="delete-btn" onclick="deleteTask(${task.id})">
+        <i class="bi bi-x-circle"></i>
+      </span>
+    `;
+    board.appendChild(card);

```

Comment:
this is essentially modifying the dom inside a loop which causes poor performance and is a bad practice